src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -494,6 +494,13 @@ kernel_impl_params primitive_inst::get_fake_aligned_params_if_possible(kernel_im
     return updated_params;
 }
 
+bool primitive_inst::all_dependencies_cpu_impl() const {
+    // Check if all dependencies are CPU impl
+    return std::all_of(dependencies().begin(), dependencies().end(),
+            [](const std::pair<const primitive_inst*, int32_t>& dep) {
+                return (dep.first->get_impl() == nullptr || dep.first->get_impl()->is_cpu());
+            });
```

Comment:
To be conservative:
```
if constant || is_cpu --> return true
otherwise --> return false
```

To be accurate
```
if constant || is_cpu --> return true
if opt_out --> return all_dependencies_cpu_impl(opt_out_code)
otherwise --> return false
```